### ISSUE TITLE
Clarify documentation for FormManager

### ISSUE THREAD

**Issue by MatthewVaadin** [vaadin/collaboration-kit#84]
The Javadoc for the `FormManager` constructor has the following text:
```
/**
     * Creates a new manager for the provided component.
     * <p>
     * The provided user information is used to set the highlight for the local
     * user with {@link #highlight(String,boolean)} or
     * {@link #highlight(String,boolean,int)} (the default is {@code false}).
     *
     * @param component
     *            the component which holds UI access, not {@code null}
     * @param localUser
     *            the information of the local user, not {@code null}
     * @param topicId
     *            the id of the topic to connect to, not {@code null}
     */
```
The `component` parameter is only described with "the component which holds UI access, not {@code null}". This does not explain that the component must be attached to the UI for the Form Manager to function as expected.

In addition, the docs site does not explain this either. There are examples where a layout component is used, which may not be enough for the developer to infer what components can (or should) be used.

In the Connection Context documentation, the effect of a component being attached or not is explained. This page could be linked for those who want to understand the lower-level workings of the kit, but the Form Manager documentation should give some advice on which component to use for the expected use cases.

**Comment by enver-haase** [comment]
Yes, and maybe hint to simply using the UI component itself (and point out any negative consequences, if any).
